storage/aptosdb/src/lib.rs

```diff
@@ -1157,12 +1153,9 @@ impl DbReader for AptosDB {
         version: Version,
     ) -> Result<Option<StateValue>> {
         gauged_api("get_state_value_by_version", || {
-            error_if_version_is_pruned(
-                &self.pruner,
-                PrunerIndex::StateStorePrunerIndex,
-                "State",
-                version,
-            )?;
+            if let Some(state_pruner) = &self.state_pruner {
```

Comment:
point

Reply:
Discussed offline. This is not introduced from the refactor and will be addressed in https://linear.app/aptoslabs/issue/BLK-82/%5Bdb%5D-state-kvs-prune-with-the-ledger-not-the-state-tree.